### ISSUE TITLE
Docs: order differs between Options and defaultOptions

### ISSUE THREAD

**Issue by BBosman** [typescript-eslint/typescript-eslint#8477]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

If you look at the documentation for the [consistent-type-imports](https://typescript-eslint.io/rules/consistent-type-imports/) rule the Options are given as this:
```ts
type Options = [
  {
    disallowTypeAnnotations?: boolean;
    fixStyle?: 'inline-type-imports' | 'separate-type-imports';
    prefer?: 'no-type-imports' | 'type-imports';
  },
];

const defaultOptions: Options = [
  {
    prefer: 'type-imports',
    disallowTypeAnnotations: true,
    fixStyle: 'separate-type-imports',
  },
];
```

As you can see the order of the properties in the `defaultOptions` differs from the ones in `Options`. While semantically not invalid I found it somewhat confusing.

I did not check any other rules for similar issues.

### Affected URL(s)

https://typescript-eslint.io/rules/consistent-type-imports/

**Comment by JoshuaKGoldberg** [comment]
Oh I love this issue so much, thank you for filing 😄! I'm all for consistent organization. I'd be very up for a lint rule enforcing that order.

Will ping @typescript-eslint/triage-team because I'm somewhat more maniacal than average on sorting.

**Comment by bradzacher** [comment]
if i had my way everything would be sorted alphabetically

**Comment by JoshuaKGoldberg** [comment]
Oh, would that include enabling https://eslint-plugin-perfectionist.azat.io/configs/recommended-natural?

**Comment by bradzacher** [comment]
I'd have to double check the rules but more than likely, yeah.
Autofixed alpha sorting >>> manual almost all the time.

**Comment by JoshuaKGoldberg** [comment]
Swell, filed #8479 to track enabling the lint rule.

Marking this issue as accepting PRs to spot fix specifically the `consistent-type-import` discrepancy. Thanks!

**Comment by BBosman** [comment]
I did some more digging.

Looking at the [rule sourcecode](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/consistent-type-imports.ts) they are actually in the same order (but not alphabetical):
```ts
  name: 'consistent-type-imports',
  meta: {
    schema: [
      {
        type: 'object',
        properties: {
          prefer: {
            type: 'string',
            enum: ['type-imports', 'no-type-imports'],
          },
          disallowTypeAnnotations: {
            type: 'boolean',
          },
          fixStyle: {
            type: 'string',
            enum: ['separate-type-imports', 'inline-type-imports'],
          },
        },
        additionalProperties: false,
      },
    ],
  },
  defaultOptions: [
    {
      prefer: 'type-imports',
      disallowTypeAnnotations: true,
      fixStyle: 'separate-type-imports',
    },
  ],
```

The `defaultOptions` mimic the order seen on the website, [so it appears](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/website/plugins/generated-rule-docs/insertions/insertNewRuleReferences.ts#L120) the `compile` function from  `@typescript-eslint/rule-schema-to-typescript-types` is somehow changing the order along the way.
